File: android/app/src/main/java/com/xreader/app/data/model/Chapter.kt
package com.xreader.app.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ChapterResponse(
    val id: Int,
    @SerialName("book_id") val bookId: Int,
    @SerialName("chapter_number") val chapterNumber: Int,
    val title: String? = null,
    @SerialName("text_content") val textContent: String? = null,
    @SerialName("word_count") val wordCount: Int,
    @SerialName("audio_path") val audioPath: String? = null,
    @SerialName("audio_duration") val audioDuration: Double? = null,
    val status: String,
    @SerialName("created_at") val createdAt: String,
    @SerialName("updated_at") val updatedAt: String
)

File: ios/xReader/Models/Chapter.swift
import Foundation

struct ChapterResponse: Codable, Identifiable {
    let id: Int
    let book_id: Int
    let chapter_number: Int
    let title: String?
    let text_content: String?
    let word_count: Int
    let audio_path: String?
    let audio_duration: Double?
    let status: String
    let created_at: String
    let updated_at: String
}

struct ChapterUpdate: Codable {
    let title: String?
    let text_content: String?
}
